### PR TITLE
refactored app-server internals

### LINKED ISSUE #52: setting "verify: true" is causing alexa app to timeout from echo devices
When I set the flag for verify to be true it works when I try to check my endpoint from a non-echo device.  It properly returns a 401 error message:

```
{
  "status": "failure",
  "reason": "The signaturecertchainurl HTTP request header is invalid!"
}
```

However, when I try to test the app from the Amazon Alexa dev website or any echo devices, it throws an error that the requested skill has taken too long to respond.

Switching the flag back to false or removing completely and it goes back to working from echo devices but obviously it is going to fail certification.

### PR DESCRIPTION
**Pull request by mreinstein**
this shouldn't change the public api at all.

fixes #52 & #21,  and *partially* solves #35 

I've deployed this to a test amazon skill and verified that at least the hello_world skill works.

this breaks several tests though. unfortunately I don't have time to update the tests and do the administrative work on this right now. sending this PR so others can review, and pick up where I left off.

Besides the tests and administrative duties, this should be good to go.

### CODE REVIEW

**Comment by mreinstein** [comment]
@tejashah88 @rickwargo @dblock @cpup22

**Comment by mreinstein** [comment]
also important to note: this PR depends on this landing first: https://github.com/alexa-js/alexa-app/pull/144

**Comment by dblock** [comment]
I merged #144, I'll try to find time for it.

**Comment by rickwargo** [comment]
I'll review by tomorrow.

File: package.json [hunk 1]
```diff
@@ -19,8 +19,7 @@
   },
   "license": "MIT",
   "dependencies": {
-    "alexa-app": "^2.4.0",
-    "alexa-verifier-middleware": "^0.1.8",
+    "alexa-app": "git+https://github.com/mreinstein/alexa-app.git#352f17cf9b8f541d8f32a385bfff61877b60134e",
```
Comment (dblock) [inline package.json]: I wouldn't do this, at the very least point it to "alexa-js/alexa-app".

Comment (mreinstein) [inline package.json]: it was a necesary evil while developing/testing. My deploy environment caches deps aggressively. This absolutely should point at canonical npm version as soon as possible. same goes for entries in `examples/apps/*package.json`

**Comment by cpup22** [comment]
Patiently waiting for this to get pulled in.  I have a skill under awaiting certification that I know is going to fail because of the verification piece is not properly working.  Hoping to swap this out on my backend before they get to that test (a.k.a. before I fail certification) so I don't have to wait another week.  So... let me know if you all need help testing this. ;)

**Comment by dblock** [comment]
I have a merged version in https://github.com/alexa-js/alexa-app-server/pull/56, with a bunch of failing tests. If anyone has quick test fixes, add a comment to that - I'll take a look soon.

**Comment by tejashah88** [comment]
Closing this PR in favor of #56

**Comment by dblock** [comment]
This PR breaks the debugger completely. The test failures are legit, see #56 for some of the fixes. I was able to fix template rendering, but not POST of actual intent data.

**Pull request closed without merging**